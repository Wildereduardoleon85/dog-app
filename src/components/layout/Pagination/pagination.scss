.pagination {
  padding: 10px;
  margin-top: 30px;
  ul {
    display: flex;
    max-width: fit-content;
    margin: 0 auto;
    li {
      button {
        background-color: var(--light);
        border: 1px solid lightgrey;
        margin: 0 5px;
        padding: 5px 10px;
        border-radius: 5px;
        cursor: pointer;
      }
      button:hover {
        background-color: var(--light-darken1);
        transition: all 0.3s ease;
        color: var(--primary);
      }
      button:active,
      .active:active {
        background-color: var(--primary-lighten2);
        transition: all 0.3s ease;
      }
      button:disabled:hover {
        background-color: white;
        color: rgba(16, 16, 16, 0.3);
        cursor: unset;
      }
      .active {
        background-color: var(--primary);
        color: white;
        border: 1px solid var(--primary);
      }
      .active:hover {
        background-color: var(--primary-darken1);
        color: white;
      }
    }
  }
}

@media screen and (max-width: 400px) {
  .pagination {
    ul {
      li {
        button {
          margin: 0 2px;
        }
      }
    }
  }
}
