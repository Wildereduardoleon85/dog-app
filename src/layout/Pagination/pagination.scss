.pagination {
  padding: 10px;
  ul {
    display: flex;
    li {
      button {
        background-color: var(--light);
        border: none;
        margin: 0 5px;
        padding: 3px;
      }
      .active {
        background-color: var(--primary);
      }
    }
  }
}
