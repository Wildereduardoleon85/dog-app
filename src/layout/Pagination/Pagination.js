import React from 'react';
import './pagination.scss';

const Pagination = (props) => {

    const {
        totalImages, 
        imagesPerPage, 
        maxPageNumberLimit, 
        minPageNumberLimit, 
        handlePag, 
        currentPage, 
        handleNext,
        handlePrev
    } = props

    const pageNumbers = [];

    for(let i = 1; i <= Math.ceil(totalImages / imagesPerPage); i++){
        pageNumbers.push(i)
    }

    return (
        <nav className="pagination">
            <ul>
                <li>
                    <button 
                        onClick={handlePrev}
                        disabled={currentPage === pageNumbers[0] ? true : false}
                    >
                        prev
                    </button>
                </li>
            {pageNumbers.map(i => {
                if(i < maxPageNumberLimit + 1 && i > minPageNumberLimit){
                    return (
                        <li key={i}>
                            <button 
                                id={i} 
                                onClick={handlePag} 
                                className={currentPage === i ? 'active' : null}
                            >
                                {i}
                            </button>
                        </li>
                        )
                }else{
                    return null
                }
            })}
                <li>
                    <button 
                        onClick={handleNext}
                        disabled={currentPage === pageNumbers[pageNumbers.length - 1] ? true : false}
                    >
                        next
                    </button>
                </li>
            </ul>
        </nav>
    )
}

export default Pagination
