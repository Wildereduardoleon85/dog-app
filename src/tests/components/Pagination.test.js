import React from 'react';
import { shallow } from "enzyme";
import Pagination from '../../components/layout/Pagination/Pagination';


describe('tests in <Pagination/>', () => {
    const   totalImages = 10,
            imagesPerPage = 4,
            maxPageNumberLimit = 5,
            minPageNumberLimit = 8,
            handlePag = () => {
                console.log('hello world')
            },
            currentPage = 7,
            handleNext = () => {
                console.log('Hello wordl again')
            },
            handlePrev = () => {
                console.log('handlePrev')
            }


    const wrapper = shallow( 
        <Pagination 
            totalImages={totalImages} 
            imagesPerPage={imagesPerPage}
            maxPageNumberLimit={maxPageNumberLimit}
            minPageNumberLimit={minPageNumberLimit}
            handlePag={handlePag}
            currentPage={currentPage}
            handleNext={handleNext}
            handlePrev={handlePrev}
        />
    )


    test('Should show the component correctly', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('button tag must have "Prev" as Text Content', () => {
        const btn = wrapper.find('#prevButton');
        expect(btn.text().trim()).toBe('Prev')
    })
})