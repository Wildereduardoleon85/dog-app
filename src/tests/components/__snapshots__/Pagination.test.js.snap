// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <Pagination/> Should show the component correctly 1`] = `
<nav
  className="pagination"
>
  <ul>
    <li>
      <button
        disabled={false}
        id="prevButton"
        onClick={[Function]}
      >
        Prev
      </button>
    </li>
    <li>
      <button
        disabled={false}
        onClick={[Function]}
      >
        Next
      </button>
    </li>
  </ul>
</nav>
`;
